### PR TITLE
item.obj is again a bound method on TestCase function items

### LINKED ISSUE #5390: pytest 4.2.0 unittest.TestCase can no longer access parent instance
In pytest==4.1.1 and earlier, `unittest.TestCase` functions could access their parent class instance. In pytest==4.2.0 the behavior changed. 

Was this intended? This causes some problems on my end if so and I'm looking for a work-around going foward.



```python

# test_foo.py
import unittest


class Test:
    def test_method(self):
        pass

def test_function():
    pass


class Broken(unittest.TestCase):
    def test_method(self):
        pass

# conftest.py
def pytest_runtest_makereport(item, call):
    if call.when == 'call':
        print(item.nodeid, 'obj self:', getattr(item.obj, '__self__', None))

```


```bash
$ pip install pytest==4.1.1
$ pytest -q
testing/test_foo.py::Test::test_method obj self: <test_foo.Test object at 0x7f8d1d337518>
.testing/test_foo.py::test_function obj self: None
.testing/test_foo.py::Broken::test_method obj self: test_method (test_foo.Broken)
.                                                                                                                                                                                                                                  [100%]
3 passed in 0.05 seconds
```


```bash
$ pip install pytest==4.2.0
$ pytest -q
testing/test_foo.py::Test::test_method obj self: <test_foo.Test object at 0x7fc6de6aff28>
.testing/test_foo.py::test_function obj self: None
.testing/test_foo.py::Broken::test_method obj self: None .     <======== this is None!
.                                                                                                                                                                                                                                  [100%]
3 passed in 0.07 seconds
```

### PR DESCRIPTION
**Pull request by nicoddemus**
Fix #5390

### CODE REVIEW

File: changelog/5390.bugfix.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Fix regression where the ``obj`` attribute of ``TestCase`` items was no longer bound methods.
```
Comment (blueyed) [inline changelog/5390.bugfix.rst]:
```suggestion
Fix regression where the ``obj`` attribute of ``TestCase`` items was no longer bound to methods.
```

File: testing/test_unittest.py [hunk 2]
```diff
@@ -139,6 +139,29 @@ def test_func2(self):
...
+        """
+        import unittest
+
+        class Test(unittest.TestCase):
+            def test_foo(self):
+                pass
+    """
+    )
+    result = testdir.runpytest_inprocess()
+    result.stdout.fnmatch_lines(["*1 passed*"])
```
Comment (blueyed) [inline testing/test_unittest.py]:
```suggestion
    result.stdout.fnmatch_lines(["* 1 passed in*"])
```

Ensures there are no warnings.

File: changelog/5390.bugfix.rst (hunk 1 above)
Comment (blueyed) [inline changelog/5390.bugfix.rst]:
```suggestion
Fix regression where the ``obj`` attribute of ``TestCase`` items was no longer a bound method.
```

File: testing/test_unittest.py (hunk 2 above)
Comment (nicoddemus) [reply testing/test_unittest.py]: Done, thanks

File: changelog/5390.bugfix.rst (hunk 1 above)
Comment (nicoddemus) [reply changelog/5390.bugfix.rst]: Done, thanks

**Comment by asottile** [comment]
backport: #5399

**Pull request merged**